app_dart/lib/src/request_handling/cache_request_handler.dart

```diff
@@ -56,7 +56,7 @@ class CacheRequestHandler<T extends Body> extends RequestHandler<T> {
     final Uint8List? cachedResponse = await cache.getOrCreate(
       responseSubcacheName,
       responseKey,
-      createFn: () => getBodyBytesFromDelegate(delegate),
+      createFn: () async { return await getBodyBytesFromDelegate(delegate); },
```

Comment:
In this example, the semantic meaning is the same, the anonymous function we're creating will immediately invoke `getBodyBytesFromDelegate()`, which will immediately return a future, and then the anonymous function will immediately return that Future, which will be completed at some time in the future.

Reply:
Okay so it does not need the explicit async and await keywords and the code that calls that function will allow the future to complete if it awaits the call?

Basically this is doing nothing correct?

Reply:
Correct. The issue appears to be related directly to the package:neat_cache, and you may want to send more debugging PRs there.

I suspect you could write an integration test that spawns a redis instance, and spawn a 1000 threads to make operations against it to reproduce this.

Reply:
Cool! Thanks for clarifying. It seems that this might not be totally thread-safe.